### PR TITLE
WEBEU2 - Redux-Swapi - Sulaiman Abidemi 🇳🇬

### PR DESCRIPTION
**Pull request by abidex4yemi**
###What does this PR do?

* Create Redux-Swapi app using React and Redux.

### CODE REVIEW

File: src/components/Character.js [hunk 1]
```diff
@@ -1,7 +1,24 @@
-import React from "react";
+import React from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
 
-const Character = props => {
-  return <li>{props.character.name}</li>;
+const Character = ({ character }) => {
```
Comment (maximesalomon) [inline src/components/Character.js:5]: Nice destructuring!

File: src/components/Character.js [hunk 2]
```diff
@@ -1,7 +1,24 @@
...
+	return <CharacterStyled>{character.name}</CharacterStyled>;
 };
 
 export default Character;
+
+Character.propTypes = {
```
Comment (maximesalomon) [inline src/components/Character.js:11]: Continue to use PropTypes. That's awesome :)

File: src/components/Character.js [hunk 3]
```diff
@@ -1,7 +1,24 @@
...
+	character: PropTypes.object.isRequired
+};
+
+const CharacterStyled = styled.li`
```
Comment (maximesalomon) [inline src/components/Character.js:15]: You put the styling with styled-components below your component logic. Awesome! Way easier for me and for you to read :)

File: src/redux/store/index.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { createStore, applyMiddleware, compose } from 'redux';
+import thunk from 'redux-thunk';
+import rootReducer from '../reducers';
+
+export const store = createStore(
+	rootReducer,
```
Comment (maximesalomon) [inline src/redux/store/index.js:6]: Awesome naming it rootReducer!

File: src/views/CharacterListView.js [hunk 5]
```diff
@@ -1,35 +1,48 @@
...
+import Loader from 'react-loader-spinner';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { CharacterList } from '../components';
+import { fetchCharacters } from '../redux/actions';
 
-import { CharacterList } from "../components";
-// import actions
+const CharacterListView = props => {
+	const { characters, fetching, fetchCharacters } = props;
```
Comment (maximesalomon) [inline src/views/CharacterListView.js:10]: Great job on the destructuring!

File: src/views/CharacterListView.js [hunk 6]
```diff
@@ -1,35 +1,48 @@
...
+	}, []);
 
-  componentDidMount() {
-    // call our action
-  }
+	return (
+		<CharacterListViewStyled>
+			{fetching ? (
+				<div>
+					<Loader type="Circles" color="#7d5bbe" height="100" width="100" />{' '}
```
Comment (maximesalomon) [inline src/views/CharacterListView.js:20]: Fancy 💅

**Review by maximesalomon — COMMENTED** [review]
Hello Yemi 🔥 

Incredible job on this project! I've added a few comments to improve your code.

**Comment by abidex4yemi** [comment]
> Hello Yemi 🔥
> 
> Incredible job on this project! I've added a few comments to improve your code.

Thank you so much

**Pull request merged**